requirements.txt

```diff
@@ -1,14 +1,14 @@
-click ~= 8.1
-ipython >= 7, < 9
-ipdb >= 0.13, < 1.0
-jinja2 ~= 3.1.2
-jsonschema ~= 4.17.3
-natsort >= 5, < 9
-netcdf4 >= 1.2.2, < 1.7
-numpy ~= 1.23.5
-pandas ~= 1.5.2
-pyomo ~= 6.4.4
-pyparsing ~= 3.0.9
-ruamel.yaml ~= 0.17.21
-scikit-learn ~= 1.2.0
-xarray ~= 2022.3.0
+click >= 8, < 9
+ipython >= 8, < 9
+ipdb >= 0.13, < 0.14
+jinja2 >= 3, < 4
+jsonschema >= 4.17, < 4.19
+natsort >= 8, < 9
+netcdf4 >= 1.2, < 1.7
+numpy >= 1, < 2
+pandas >= 2, < 3
+pyomo >= 6.5, < 7
+pyparsing >= 3.0, < 3.1
+ruamel.yaml >= 0.17, < 0.18
+scikit-learn >= 1.2, < 1.3
+xarray >= 2023, < 2024
```

Comment:
Not sure this is a good idea, why not keep pinning versions of key dependencies more explicitly?

Reply:
Some flexibility allows for calliope to be installed more easily alongside other packages. Between major releases of mature dependencies, there should be no breaking changes to be concerned with. Between younger packages, allowing only changes in patch releases seems fair. We could allow only changes between patch releases for everything if you prefer, but I would allow some of this flexibility.